src/main/ruby/truffleruby/core/time.rb

```diff
@@ -336,7 +336,13 @@ def at(sec, usec=undefined, unit=undefined, **kwargs)
           raise ArgumentError, "unexpected unit: #{unit}"
         end
 
-      usec = 0 if Primitive.undefined?(usec)
+      if Primitive.undefined?(usec) && Primitive.object_kind_of?(sec, Rational)
+        sec, fractional_seconds = sec.divmod(1)
+        micro_fractional_seconds = fractional_seconds * 1_000_000
+        usec = micro_fractional_seconds.to_i
```

Comment:
This won't work if the Rational has non-zero nanoseconds though, right?
Also, `usec` will be impacted by `second_arg_scale` which seems incorrect.

Would it work to fix the computation below by using `divmod(1)` instead of `to_i` + `% 1.0`?

Reply:
I updated the fix and added an early return for the Rational case. In the case where `Rational` is used, there is no need for the logic (like `second_arg_scale`) following the if-block. 

This fix is by no means elegant, but I couldn't see a better way to integrate the fix without refactoring the entire method. What do you think?